tests/db_functions/json/test_json_set.py

```diff
@@ -0,0 +1,133 @@
+from django.db.models.functions.json import JSONSet
+from django.test import TestCase
+
+from ..models import UserPreference
+
+
+class JSONSetTests(TestCase):
```

Comment:
I don't know if we'll need a new feature flag, but at the very least, these tests need to check for `supports_json_field` for now

```suggestion
@skipUnlessDBFeature("supports_json_field")
class JSONSetTests(TestCase):
```